en/salesforce/src/docs/TEMPLATE_ms_composer_custom_expression_editor.adoc

```diff
@@ -1,85 +1,304 @@
-* The data pills and formulas displayed in the if/else block for the value are dependent on the data type of the selected field.
+//More content may be added - to be determined later
+* You can nest functions. You can use the input parameters of another function call for your function.
+* Composer functions used in Composer are built using https://docs.mulesoft.com/dataweave/2.3/dw-operators[DataWeave].
 
-== DataWeave Functions
+=== String Functions
+
+The *Custom Expression Editor* supports the following `string` formulas and operators:
+
+[%header%autowidth.spread]
+|===
+
+|Function Name |Description |Example |Input Data Type Parameters |Output Data Type
+
+|lower | Returns the provided string in lowercase characters. |`lower(COMPOSER) = composer` |text: String |String
+
+|upper |Returns the provided string in uppercase characters. |`upper(composer) = COMPOSER` |text: String |String
+
+|capitalize |Capitalizes the first letter of each word in a string. This also removes underscores between words and puts a space before each capitalized word.
+|`capitalize(name) = Name` +
+
+`capitalize(firstName) = First Name` +
+
+`capitalize(customer_first_name) = Customer First Name` +
+|text: String |String
+
+|trim |Removes any blank spaces from the beginning and end of a string.
+|`trim(<long example text>) = this example long text` |text: String |String
+
+|first |Returns the specified number of characters from the beginning of a string. If the number is equal to or greater than the number of characters in the string, the function returns the entire string. |`first(Hello Composer!, 5) = Hello` |text: String +
+
+numCharacters: Number |String
+
+|last |Returns the specified number of characters from the end of a string. |`last(Hello Composer!, 9) = Composer!` |text: String +
+
+numCharacters: Number|String
+
+|substitute |Replaces all substrings that match a literal search string with a specified replacement string. Replacement proceeds from the beginning of the string to the end. For example, the result of replacing `aa` with `b` in the string `aaa` is `ba`, rather than `ab`. |`substitute(Composer, o, O) = COmpOser` +
+
+`substitute(Composer, a, Example) = Composer`
+
+|text: String +
+
+target: String +
+
+replacement: String |String
+
+|===
+
+It is important to note that in a `string` field, you can use `number` expressions selected from the operator list; however, the expressions must be enclosed within parentheses. For example, you can enter `abc(9+9)` where the `+` is selected from the operator list. This string renders as `abc18`.
+
+=== Number Functions
+
+In the editor, aa mathematical operator (+,-,*, or /) selected from the operator list is different from an operator typed and resolved as a string character. For example, the entry `9+9` is considered a string rather than `18`. To use math operators, select them from the operator list or use xref:ms_composer_custom_expression_editor.adoc#keyboard-shortcuts[keyboard shortcuts].
```

Comment:
```suggestion
In the editor, a mathematical operator (+,-,*, or /) selected from the operator list is different from an operator typed and resolved as a string character. For example, the entry `9+9` is considered a string rather than `18`. To use mathematical operators, select them from the operator list or use xref:ms_composer_custom_expression_editor.adoc#keyboard-shortcuts[keyboard shortcuts].
```